integration/query_comparison_compat_test.go

```diff
@@ -452,10 +452,8 @@ func testQueryComparisonCompatGt() map[string]queryCompatTestCase {
 						{"array", bson.A{int32(42), "foo", nil}}, {"42", "foo"}, {"foo", int32(42)},
 					}},
 				}},
-				{"_id", bson.D{{"$ne", "array-documents-nested"}}}, // satisfies the $gt condition
 			},
-			resultType:     emptyResult,
-			resultPushdown: true,
+			resultPushdown: false,
```

Comment:
@noisersup I need your opinion regarding this one. I added one more value to shareddata composites - `array-documents-nested-duplicate`. I needed one more value to test that sort works correctly for dot notation (as I need more than one document). After that, this test started failing. I'm not sure why exactly `array-documents-nested` was excluded (I guess, to only work with "simpler" documents, not this one). I included it back, and after that, the test started working without pushdown (so I set `resultPushdown` as `false`). Alternatively, I can exclude both `array-documents-nested` and `array-documents-nested-duplicate` from this test. Still, such exclusion makes the test a bit more complicated, so I wonder if the current approach is good enough from the pushdown point of view. Is it correct that it returns `false`?

Reply:
It's because this PR removes line 455 with the `$ne` operator, so we don't perform any pushdown here, as we don't support `$gt` yet

Reply:
It's okay to remove it! Everything that covers `$eq` should be inside `Eq` tests, and in this test case we do pushdown "by accident"